internal/agent/device/lifecycle/manager.go

```diff
@@ -92,15 +96,71 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.log.Warnf("Detected decommissioning request from flightctl service")
+		errored := false
```

Comment:
nit: we should use actual errors as conditions of error

Reply:
Yeah, I guess once we have more handling here we can just return early if an error comes up so we won't need this bool

Reply:
I would say we don't need it now right it can't be true right?